tests/integration/features/Shibboleth.feature

```diff
@@ -165,6 +165,139 @@ Feature: Shibboleth
+    And I should be redirected to "https://localhost:4443/idp/profile/SAML2/Redirect/SSO"
+    And I send a POST request to "https://localhost:4443/idp/profile/SAML2/Redirect/SSO?execution=e1s1" with the following data
+      |j_username|j_password|_eventId_proceed|
+      |staff1    |password  |                |
+    And The response should be a SAML redirect page that gets submitted
+    And I send a GET request with requesttoken to "http://localhost:8080/index.php/apps/user_saml/saml/sls"
+    And the cookies are cleared
+    # Have a third SAML users being known
+    And I send a GET request to "http://localhost:8080/index.php/login"
+    And I should be redirected to "https://localhost:4443/idp/profile/SAML2/Redirect/SSO"
+    And I send a POST request to "https://localhost:4443/idp/profile/SAML2/Redirect/SSO?execution=e1s1" with the following data
+      |j_username|j_password|_eventId_proceed|
+      |student2  |password  |                |
+    And The response should be a SAML redirect page that gets submitted
+    And I send a GET request with requesttoken to "http://localhost:8080/index.php/apps/user_saml/saml/sls"
+    And the cookies are cleared
+    # The users are already known as members - student 2 is not
+    And the user "student1" is added to the group "Astrophysics"
+    And the user "staff1" is added to the group "Astrophysics"
+    # Set the proper attributes
+    And The setting "saml-attribute-mapping-email_mapping" is set to "urn:oid:0.9.2342.19200300.100.1.3"
+    And The setting "saml-attribute-mapping-displayName_mapping" is set to "urn:oid:2.5.4.42 urn:oid:2.5.4.4"
+    And The setting "saml-attribute-mapping-group_mapping" is set to "groups"
+    # Now login in for good
+    When I send a GET request to "http://localhost:8080/index.php/login"
+    Then I should be redirected to "https://localhost:4443/idp/profile/SAML2/Redirect/SSO"
+    And I send a POST request to "https://localhost:4443/idp/profile/SAML2/Redirect/SSO?execution=e1s1" with the following data
+      |j_username|j_password|_eventId_proceed|
+      |student1  |password  |                |
+    And The response should be a SAML redirect page that gets submitted
+    And The user value "groups" should be "Astrophysics,SAML_Students"
+    And the group backend of "Astrophysics" should be "Database"
+    And Then the group backend of "Astrophysics" must not be "user_saml"
+    And I execute the background job for class "OCA\\User_SAML\\Jobs\\MigrateGroups"
+    And the group backend of "Astrophysics" should be "user_saml"
+    And Then the group backend of "Astrophysics" must not be "Database"
+    # Recreate state from v6.1 where members where not migrated - terribly awful tinkering
+    And I "disable" the app "user_saml"
+    And the local group "Astrophysics" is created
+    And I hack "student2" into existence
+    And the user "student2" is added to the group "Astrophysics"
+    And I "enable" the app "user_saml"
```

Comment:
This is plain awful, but I did not come up with any other way to resemble this situation. Easier would be issuing direct SQL queries, however this would be written again and support all DB flavours, and detect what is running on which port.